mico-core/src/main/java/io/github/ust/mico/core/service/MicoKubernetesClient.java

```diff
@@ -1253,4 +1253,81 @@ public PasswordAuthentication getOpenFaasCredentials() {
         return new PasswordAuthentication(userName, password.toCharArray());
     }
 
+
+    /**
+     * Requests the public IP of a Kubernetes service and returns it or an empty {@code Optional} if the
+     * service has no public IP.
+     *
+     * @param name      the name of the service.
+     * @param namespace the namespace which contains the service.
+     * @return the public ip of a service or an empty optional.
+     * @throws KubernetesResourceException if there is no such service.
+     */
+    public Optional<String> getPublicIpOfKubernetesService(String name, String namespace) throws KubernetesResourceException {
+        log.debug("Requesting public ip of Kubernetes service '{}' in namespace '{}'", name, namespace);
+        Service service = getServiceOrThrowKubernetesResourceException(name, namespace);
+        List<LoadBalancerIngress> ingresses;
+        ServiceStatus status = service.getStatus();
+        if (status == null || status.getLoadBalancer() == null || status.getLoadBalancer().getIngress() == null) {
+            throw new KubernetesResourceException("The Kubernetes service with the name '" + name + "' in the namespace '" + namespace + "' does not" +
+                " contain a status, loadbalancer or ingress element.");
+        }
+        ingresses = service.getStatus().getLoadBalancer().getIngress();
+        if (ingresses.size() != 1) {
+            log.debug("There is no or more than one ingress. There should be only one.");
```

Comment:
Why not just log a warning and return the first ip if there are two or more?

Reply:
That's actually the case in the mentioned clone:
https://github.com/UST-MICO/mico/blob/c7ac6725b173b76b41111996f2cd1e0c0a5e83cc/mico-core/src/main/java/io/github/ust/mico/core/service/MicoStatusService.java#L279

Reply:
I think the semantic is not clear if there is more than one, therefore I think it is better to report an error. Simply returning the first could mask an error. As we discussed this case should not happen and it indicates an error in our configuration.